### PR TITLE
automatically start read servers

### PR DESCRIPTION
**Pull request by aletomsic**
This PR addresses issue #266.
Added an option (which defaults to true) to start read_servers automatically.
When antidote is meant to be run in a cluster, this has to be set to false, and start the read servers afterwards.
This is the best idea I could come up with that does not penalize the interactive coordinator and, therefore, all transactions.
A more definitive and automatic way of solving this issue should be envisioned (maybe by more enlightened people ;).

### CODE REVIEW

**Comment by peterzeller** [comment]
@bieniusa Can you have a look at this and merge? It would be important for testing things locally, for example using the Docker image.

File: src/antidote_app.erl [hunk 1]
```diff
@@ -65,6 +65,13 @@ start(_StartType, _StartArgs) ->
                   ok
             end,
 
+            case application:get_env(antidote, auto_start_read_servers) of
+                {ok, true} ->
+                    %% start read servers
+                    inter_dc_manager:start_bg_processes(stable);
+                _->
+                    dont_start_read_servers
```
Comment (bieniusa) [inline src/antidote_app.erl]: Why does this return an atom??

File: src/antidote.app.src [hunk 2]
```diff
@@ -39,12 +39,15 @@
   %%                stored to disk (this is very slow in the current logging setup)
   %%        false -> all updates are sent to the operating system to be stored to disk (eventually), but are not guaranteed to be stored durably on disk
   %%                 when the reply is sent
+  %% %% auto_start_read_servers:
+  %%        true -> read servers will start automatically. It should be set to true when Antidote will be run in a single node/machine.
+  %%        false -> read servers will not start automatically. It should be set to false when Antidote will be run in a cluster, and later on start them.
```
Comment (bieniusa) [inline src/antidote.app.src]: This sentence I do not understand "and later on start them".

**Comment by peterzeller** [comment]
> Can we have a systest that tests this code?

It is hard to test with our system tests. I created another pull request with an alternative way to test it ( #270 )

File: src/antidote_app.erl (hunk 1 above)
Comment (aletomsic) [inline src/antidote_app.erl]: this atom is not assigned to anything nor returned.
It is there just to "explain" that read servers won't be started.
In this branch of the case, nothing should be done.
Should I use an IF instead?

**Pull request merged**